.github/workflows/ci-spell-checking.yml

```diff
@@ -0,0 +1,44 @@
+name: ci-spell-checking
+
+on:
+  push:
+    paths-ignore:
+      - '.vscode/styles/**'
+    branches: 
+    - main
+  pull_request:
+    types:
+    - opened
+    - reopened
+    - synchronize
+    - ready_for_review
+    branches: 
+    - main
+
+jobs:
+  spell-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Get changed files in the content folder
+        id: changed-files-specific
+        uses: tj-actions/changed-files@v35
+        with:
+          files: ./
+          files_ignore: './.vscode/'
+      - name: Install Vale
+        if: steps.changed-files-specific.outputs.any_changed == 'true'
+        run: |
+          sudo apt-get update
+          sudo apt-get install curl

```

Comment:
we should avoid sudo commands. And it seems reasonable to assume curl is present.